object-api-multipart.go

```diff
@@ -243,14 +242,17 @@ func (o objectAPI) ListMultipartUploads(bucket, prefix, keyMarker, uploadIDMarke
 		}).Errorf("listMetaVolumeFiles failed with %s", e)
 		return ListMultipartsInfo{}, probe.NewError(e)
 	}
+
+	// Loop through all the received files fill in the multiparts result.
 	for _, fi := range fileInfos {
 		var objectName string
 		var uploadID string
 		if fi.Mode.IsDir() {
+			// All directory entries are common prefixes.
 			objectName = strings.TrimPrefix(fi.Name, retainSlash(bucket))
-			// For a directory entry
 			result.CommonPrefixes = append(result.CommonPrefixes, objectName)
-			continue
+			result.NextKeyMarker = objectName
+			result.NextUploadIDMarker = ""
```

Comment:
if you do `result.NextUploadIDMarker = uploadID` after the if/else block this will be automatically taken care of as uploadID will be `nil` if code entered `if` block and `non-nil` if the code entered `else` block,

i.e code before this commit is fine.